openedx/adg/lms/applications/tests/test_helpers.py

```diff
@@ -352,3 +367,170 @@ def test_has_admin_permissions(mocker, is_business_line_admin, is_superuser, is_
     test_user = UserFactory(is_superuser=is_superuser, is_staff=is_staff)
 
     assert has_admin_permissions(test_user) == expected_result
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'written_application, omni_courses, bu_courses, omni_courses_started, bu_courses_started, congrats, msg, inst',
+    [
+        (False, False, False, False, False, '', '', ''),
+        (
+            True, False, False, False, False, WRITTEN_APPLICATION_COMPLETION_CONGRATS,
+            WRITTEN_APPLICATION_COMPLETION_MSG, ''
+        ),
+        (True, False, False, True, False, '', '', WRITTEN_APPLICATION_COMPLETION_INSTRUCTION),
+        (
+            True, True, False, False, False, PREREQUISITE_COURSES_COMPLETION_CONGRATS,
+            PREREQUISITE_COURSES_COMPLETION_MSG, ''
+        ),
+        (True, True, False, False, True, '', '', PREREQUISITE_COURSES_COMPLETION_INSTRUCTION),
+        (True, True, True, False, False, APPLICATION_SUBMISSION_CONGRATS, APPLICATION_SUBMISSION_INSTRUCTION, ''),
+    ],
+    ids=[
+        'no_step_completed', 'written_app_completed', 'omni_courses_started',
+        'omni_courses_completed', 'bu_courses_started', 'all_steps_completed'
+    ]
+)
+def test_get_application_hub_instructions(
+    written_application, omni_courses, bu_courses, omni_courses_started, bu_courses_started, congrats, msg, inst
+):
+    """
+    Test the congratulation messages and instructions for each requirement completion till application submission.
+    """
+    user = UserFactory(username=USERNAME, password=PASSWORD)
+    application_hub = ApplicationHubFactory(user=user)
+
+    if written_application:
+        application_hub.submit_written_application_for_current_date()
+    if omni_courses:
+        application_hub.set_is_prerequisite_courses_passed()
+    if bu_courses:
+        application_hub.is_bu_prerequisite_courses_passed = True
```

Comment:
Can we please add set method for `is_bu_prerequisite_courses_passed` to add more consistency?